fixtures/load-repo.js

```diff
@@ -0,0 +1,205 @@
+  console.log(kleur.bold().blue(`Cloning repo: ${name} from ${url}`));
+  // Clear out all existing repos
+  if (existsSync(reposDir)) await rm(reposDir);
+  await mkdir(reposDir);
+  await runCommand('git', ['clone', url, getRepoDir(name)]);
+  console.log(kleur.bold().blue(`Done cloning repo: ${name}`));
+};
+
+const main = async () => {
+  const { url } = await prompts({
+    name: 'url',
+    type: 'text',
+    message: 'GitHub Repo URL',
+  });
+  const name = url.replace(/\.git$/, '').replace(/.*\//, '');
+  await cloneRepo(name, url);
+  const files = await walk(name, { filesonly: true, cwd: reposDir });
+
+  console.log(kleur.bold().blue('Looking for files to lint'));
+
+  const interestingFiles = await Promise.all(
+    files
+      .filter((f) => f.endsWith('.js') && !f.replace(name, '').includes('test'))
+      .map(async (file) => {
+        const contents = await readFileAsync(path.join(reposDir, file), 'utf8');
+        return { file, contents };
+      })
+  );
+
+  const eslint = new CLIEngine({
+    configFile: path.join(fixturesDir, '.eslintrc.js'),
+    fix: true,
+  });
+
+  const filesWithNumNodeTypes = await Promise.all(
+    interestingFiles.map(async ({ file, contents }) => {
+      const seenNodeTypes = new Set();
+      let numNodes = 0;
+      let parsed;
+      /**
+       * This should be returned if the file is invalid
+       * This could happen if there is a parsing error or if it is empty
+       * or if it has too many linting errors, etc
+       */
+      const invalidResult = {
+        file,
+        contents,
+        score: 0,
+        numLintErrors: 0,
+        numNodeTypes: 0,
+        loc: 0,
+      };
+      const filenameRelative = path.join(reposDir, file);
+      try {
+        // Attempts to parse each file with babel
+        // Traverses the file and counts up the number of node types and number of nodes
+        // Files with more node types tend to be more interesting and better candidates for lint testing
+        parsed = await babel.parseAsync(contents, {
+          parserOpts: {
+            plugins: [
+              'jsx',
+              'classProperties',
+              'dynamicImport',
+              'nullishCoalescingOperator',
+              'optionalChaining',
+            ],
+          },
+          filenameRelative,
+        });
+      } catch (error) {
+        return invalidResult;
+      }
+
+      babel.traverse(parsed, {
+        enter(path) {
+          numNodes++;
+          seenNodeTypes.add(path.type);
+        },
+      });
+
+      const numNodeTypes = seenNodeTypes.size;
+
+      let eslintReport;
+      try {
+        eslintReport = eslint.executeOnText(contents).results[0];
+      } catch (error) {
+        console.log('eslint error', file, error);
+        return invalidResult;
+      }
+
+      const numLintErrors = eslintReport.errorCount + eslintReport.warningCount;
+      const eslintFixedContents = eslintReport.output || contents;
```

Comment:
Can you briefly explain why this is either the report output **_or_** the original file contents?

Reply:
`eslintReport.output` is a string containing the source code with all of eslint's auto-fixes applied. Sometimes that will be `undefined`, if there are no auto-fixes. `|| contents` lets it default to the original input code if there are no fixes.